clients/mobile/__tests__/authentication/useLogin.test.ts

```diff
@@ -1,20 +1,70 @@
-import { mockedNavigate, setRouteParams } from '../__mocks__/reactNavigation.mock'
 import { renderHook, act } from '@testing-library/react-native'
 import { useLogin } from '../../src/authentication/login/hooks/useLogin'
 import { loginUser, googleLogin, facebookLogin } from '../../src/authentication/services/authService'
+import { SCREENS } from '../../src/setup/constant/screens'
+
+const mockDispatch = jest.fn()
+const mockSetIsAuthenticated = jest.fn()
+let mockUserProfile: { bloodGroup?: string } | null = null
```

Comment:
do we need let?

Reply:
vai, we generally avoid `let`, but i think in this particular case might actually get benefit from keeping the `let` declaration.  here `mockUserProfile` variable is specifically used for testing different user profile states, with proper reset in `beforeEach` hook to prevent test pollution. Using `const` would require a more complex closure-based approach that might make our tests harder to maintain.